### PR TITLE
Fixed #25388 -- Added an option to allow disabling of migrations running during test database creation

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/25388

### CODE REVIEW

File: django/db/migrations/loader.py [hunk 1]
```diff
@@ -64,6 +64,10 @@ def load_disk(self):
         self.unmigrated_apps = set()
         self.migrated_apps = set()
         for app_config in apps.get_app_configs():
+            if (self.connection is not None and
```
Comment (MarkusH) [inline django/db/migrations/loader.py]: I'd rather see that change in `migrations_module()` so Django doesn't suddenly pick up a module somewhere else but just doesn't know about migrations for any apps.

Comment (berkerpeksag) [inline django/db/migrations/loader.py]: Agreed, but migrations_module() is a classmethod that acts like a staticmethod so it doesn't have access to self.connection. I couldn't came up with a backward compatible solution. Do you have a suggestion?

Comment (MarkusH) [inline django/db/migrations/loader.py]: As far as I can see you can instantiate the `MigrationLoader` in the questioner and writer with `connection=None` and `load=False` to achieve the same behavior there and drop the `@classmethod` from `migrations_module()`

Comment (berkerpeksag) [inline django/db/migrations/loader.py]: Done, thanks!

File: docs/ref/settings.txt [hunk 2]
```diff
@@ -710,6 +710,18 @@ The creation-order dependencies of the database. See the documentation
...
+.. setting:: TEST_MIGRATE
+
+``MIGRATE``
+^^^^^^^^^^^
+
+.. versionadded:: 1.10
+
+Default: ``True``
+
+If it is set to ``False``, Django won't use existing migrations to create
```
Comment (MarkusH) [inline docs/ref/settings.txt]: "... create __a__ test ..."

Comment (berkerpeksag) [inline docs/ref/settings.txt]: Fixed.

**Comment by MarkusH** [comment]
Thanks for your work. Merged in 157d7f1f1de4705daddebb77f21bd7097a74513d

File: tests/migrations/test_loader.py [hunk 3]
```diff
@@ -202,6 +202,23 @@ def test_marked_as_unmigrated(self):
         self.assertEqual(migration_loader.migrated_apps, set())
         self.assertEqual(migration_loader.unmigrated_apps, {'migrated_app'})
 
+    @override_settings(
+        INSTALLED_APPS=['migrations.migrations_test_apps.migrated_app'],
+    )
+    def test_disable_migrations(self):
+        connections = ConnectionHandler({
+            'default': {
+                'NAME': 'dummy',
```
Comment (timgraham) [inline tests/migrations/test_loader.py]: This leaves a "dummy" SQLite database on the filesystem after running the tests.

Comment (berkerpeksag) [inline tests/migrations/test_loader.py]: Thanks, I've opened https://github.com/django/django/pull/6326.

**Pull request closed without merging**